### PR TITLE
added graphite support.

### PR DESCRIPTION
**Pull request by hthayko**
With -g flag the forever command sends data to graphite.
Also added optimization: each successive prior call does not make a call to db to get the latest crawl anymore, as it's available from the previous prior.

### CODE REVIEW

File: src/forever.js [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
 var src = require('./program');
+var graphite = require('graphite');
+var utils = require('./lib/utility.js');
+var graphiteClient;
 
 module.exports = function(ipp, dbUrl, commander) {
   console.log("FOREVER: calling ENTER.....")
+
+  if(commander.graphite){
+    graphiteClient = graphite.createClient('plaintext://54.91.39.21:2003/');
```
Comment (wilsonianb) [inline src/forever.js]: The graphite address should be passed in as an environment variable or command line argument

**Comment by sourencho** [comment]
Could you change the version in `package.json` before you merge? 
We want to `npm publish` the crawler every time it merges so that other programs that use it (such as the api) can get the newer version.

**Comment by sourencho** [comment]
I think we should have an independent program sending metrics to graphite by reading the latest crawl from the db.
I want to record metrics for each rippled and this will slow down the crawler significantly.

The crawler will be faster because it's only doing the minimum (storing raw data) and we can get more crawls per second.

**Comment by wilsonianb** [comment]
Agreed

**Comment by sourencho** [comment]
We should keep the forever optimization though

**Comment by hthayko** [comment]
If I'm not mistaken Kanaan said crawler should send data itself.

**Comment by hthayko** [comment]
It's going to be a separate module.

**Pull request closed without merging**